include/engine/guidance/guidance_toolkit.hpp

```diff
@@ -0,0 +1,426 @@
+#ifndef OSRM_GUIDANCE_GUIDANCE_TOOLKIT_HPP_
```

Comment:
Most of this code is only ever used in the pre-processing code path and that is where it belongs. If you see mixed usage split it up.